### PR TITLE
Initial switch from Travis-CI to Github Actions.

### PR DESCRIPTION
**Pull request by enjoy-digital**
This is a first attempt to switch the repository to Github Actions since CI is broken since last December. This not yet going to the end, @mithro would you mind helping me  finishing this?

### CODE REVIEW

**Comment by mithro** [comment]
@umarcor is the expert here.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      GIT_MODE : https
+    steps:
+      - uses: actions/checkout@v2
+
+      # Checkout Repository
+      - name: Checkout
+        run: git fetch --prune --unshallow --tags
+
+      # Install Tools
+      - name: Install Tools
```
Comment (umarcor) [inline .github/workflows/ci.yml:28]: Maybe use action setup-python instead of doing it manually. See: https://github.com/ghdl/ghdl/blob/master/.github/workflows/Test.yml#L84-L87.

File: modules.ini [hunk 2]
```diff
@@ -3,14 +3,14 @@ branch = master
...
 # -------------------------------------
-[blackparrot]
-type=cpu
-human_name = BlackParrot
-src = https://github.com/black-parrot/black-parrot.git
-contents = system_verilog
-license = License :: OSI Approved :: BSD License
-license_spdx = BSD-3-Clause
-git_hash = 7eb1037637d8515a259e204117b7b1273b1c2941
+#[blackparrot]
```
Comment (umarcor) [inline modules.ini:6]: Is this change related to the migration?

**Comment by mithro** [comment]
This seems to be working now! -- Screenshot from 2021-02-03 22-46-06

File: templates/.github/workflows/publish-to-pypi.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    branches:
+      - master
+
+jobs:
+  build-n-publish:
+
+    strategy:
+      matrix:
+        python-version: [ '2.x', '3.6', '3.7', '3.8', '3.9', 'pypy-2.7', 'pypy-3.6', 'pypy-3.7' ]
+        os: [ubuntu-latest]
```
Comment (umarcor) [inline templates/.github/workflows/publish-to-pypi.yml:14]: Overall LGTM! This line might feel redundant, but I guess it's because you want to add other platforms in the future. All good!

**Comment by enjoy-digital** [comment]
Thanks @mithro, @umarcor for looking at this! @mithro feel free to merge when you think it's OK on your side.

**Comment by umarcor** [comment]
@enjoy-digital, note that @mithro added/modified a different workflow file. This PR contains two at the moment. The one you added is unmodified.

**Comment by enjoy-digital** [comment]
@umarcor: I think @mithro also forced pushed on my initial changes to fix it.

**Comment by umarcor** [comment]
@enjoy-digital, check the comments of my first review above.

**Comment by enjoy-digital** [comment]
@umarcor: Thanks for the suggestions, I'll try to apply them.

**Comment by mithro** [comment]
There are actually two workflow files;
 1) The workfile for this repository
 2) The workfile template for the repositories generated by this template.

I think this pull request is ready to merge for now and we can continue to clean it up afterwards.
